### PR TITLE
fix(datastore): remove typename from ModelMetadata

### PR DESCRIPTION
**Pull request by mikepschneider**
- [ ] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*
https://github.com/aws-amplify/amplify-android/issues/1925
https://github.com/aws-amplify/amplify-android/issues/2066

*Description of changes:*
This PR takes option 2 to remove `__typename` from ModelMetadata table, see https://github.com/aws-amplify/amplify-android/issues/2066#issuecomment-1300680387 . It essentially reverts one of the changes in the [CPK PR](https://github.com/aws-amplify/amplify-android/pull/1650) at the [file](https://github.com/aws-amplify/amplify-android/pull/1650/files#diff-5cdf2885642e1d230ebe77616604dc89e3ed4deb7975f34b69159e0affd9dc27)  (FYI @poojamat)

The changes introduced in CPK include an expectation that there exists a `__typename` field on each model table.  But when an existing customer upgrades without clearing their local storage, that assumption will fail.  This code gets rid of the expectation that `__typename` exists. 

*How did you test these changes?*

- Passed existing unit tests
- 
- [ ] Added Unit Tests
- [ ] Added Integration Tests

*Documentation update required?*
- [x ] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by drochetti — DISMISSED** [review]

**Review by manueliglesias — DISMISSED** [review]
LGTM 👍

**Comment by mikepschneider** [comment]
The integration tests are super flaky but I did get one clean run.  The other failures are almost always a single failure on one device and the other 2 succeed.  So I'm going to call that good and submit it.

File: .gitignore [hunk 1]
```diff
@@ -111,4 +111,5 @@ build-artifacts.tar.gz
 
 # sample app resources
 sample*
-.graphqlconfig.yml
\ No newline at end of file
+.graphqlconfig.yml
+aws-datastore/src/androidTest/res/raw/amplifyconfigurationupdated.json
```
Comment (tylerjroach) [inline .gitignore:115]: May not be a bad idea to move to `**/awsconfiguration.json` in the section where we exclude these similar files. Also not sure what is generating this? Not holding up approval on this though.

**Review by tylerjroach — APPROVED** [review]

**Review by manueliglesias — APPROVED** [review]
LGTM

File: .gitignore (hunk 1 above)
Comment (mikepschneider) [reply .gitignore:115]: I added that because that file kept getting adding accidentally.

**Pull request merged**